lib/recipes-data/src/lib/extract-all-recipes.test.ts

```diff
@@ -0,0 +1,201 @@
+import {AssetType} from "@guardian/content-api-models/v1/assetType";
+import type {Content} from "@guardian/content-api-models/v1/content";
+import {ContentType} from "@guardian/content-api-models/v1/contentType";
+import {ElementType} from "@guardian/content-api-models/v1/elementType";
+import {extractAllRecipesFromArticle} from "./extract-recipes";
+import {makeCapiDateTime} from "./utils";
+
+describe("extractAllRecipesFromAnArticle", () => {
+
+  it("should work if main and body both block contains recipes", () => {
```

Comment:
I'd also put in a checks to see that:
- it's OK if there are no recipes
- multiple body-block recipes passed through correctly
- a single invalid recipe (no id field) is dropped (see later comments about this - currently the code throws an exception)
- a single invalid recipe (invalid json) is dropped